core/trino-main/src/main/java/io/trino/operator/scalar/JsonOutputFunctions.java

```diff
@@ -0,0 +1,168 @@
+            JsonEncoding.UTF32_LE,
+            Charset.forName("UTF-32LE"),
+            Slices.copiedBuffer(new ArrayNode(JsonNodeFactory.instance).asText(), Charset.forName("UTF-32LE")),
+            Slices.copiedBuffer(new ObjectNode(JsonNodeFactory.instance).asText(), Charset.forName("UTF-32LE")));
+
+    private JsonOutputFunctions() {}
+
+    @SqlNullable
+    @ScalarFunction(value = "$json_to_varchar", hidden = true)
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice jsonToVarchar(@SqlType(StandardTypes.JSON_2016) JsonNode jsonExpression, @SqlType(StandardTypes.TINYINT) long errorBehavior, @SqlType(StandardTypes.BOOLEAN) boolean omitQuotes)
+    {
+        return toSlice(jsonExpression, UTF_8, errorBehavior, omitQuotes);
+    }
+
+    @SqlNullable
+    @ScalarFunction(value = "$json_to_varbinary", hidden = true)
+    @SqlType(StandardTypes.VARBINARY)
+    public static Slice jsonToVarbinary(@SqlType(StandardTypes.JSON_2016) JsonNode jsonExpression, @SqlType(StandardTypes.TINYINT) long errorBehavior, @SqlType(StandardTypes.BOOLEAN) boolean omitQuotes)
+    {
+        return jsonToVarbinaryUtf8(jsonExpression, errorBehavior, omitQuotes);
+    }
+
+    @SqlNullable
+    @ScalarFunction(value = "$json_to_varbinary_utf8", hidden = true)
+    @SqlType(StandardTypes.VARBINARY)
+    public static Slice jsonToVarbinaryUtf8(@SqlType(StandardTypes.JSON_2016) JsonNode jsonExpression, @SqlType(StandardTypes.TINYINT) long errorBehavior, @SqlType(StandardTypes.BOOLEAN) boolean omitQuotes)
+    {
+        return toSlice(jsonExpression, UTF_8, errorBehavior, omitQuotes);
+    }
+
+    @SqlNullable
+    @ScalarFunction(value = "$json_to_varbinary_utf16", hidden = true)
+    @SqlType(StandardTypes.VARBINARY)
+    public static Slice jsonToVarbinaryUtf16(@SqlType(StandardTypes.JSON_2016) JsonNode jsonExpression, @SqlType(StandardTypes.TINYINT) long errorBehavior, @SqlType(StandardTypes.BOOLEAN) boolean omitQuotes)
+    {
+        return toSlice(jsonExpression, UTF_16, errorBehavior, omitQuotes);
+    }
+
+    @SqlNullable
+    @ScalarFunction(value = "$json_to_varbinary_utf32", hidden = true)
+    @SqlType(StandardTypes.VARBINARY)
+    public static Slice jsonToVarbinaryUtf32(@SqlType(StandardTypes.JSON_2016) JsonNode jsonExpression, @SqlType(StandardTypes.TINYINT) long errorBehavior, @SqlType(StandardTypes.BOOLEAN) boolean omitQuotes)
+    {
+        return toSlice(jsonExpression, UTF_32, errorBehavior, omitQuotes);
+    }
+
+    private static Slice toSlice(JsonNode json, EncodingSpecificConstants constants, long errorBehavior, boolean omitQuotes)
+    {
+        if (omitQuotes && json.isTextual()) {
+            return Slices.copiedBuffer(json.asText(), constants.charset);
+        }
+
+        ByteArrayBuilder builder = new ByteArrayBuilder();
+        JsonGenerator generator;
+        try {
+            generator = MAPPER.createGenerator(builder, constants.jsonEncoding);
```

Comment:
generator needs to by closed
use `try(JsonGenerator generator = MAPPER.createGenerator(builder, constants.jsonEncoding) {`

Reply:
Btw - is there a way to have a pool of generators and reuse? Not sure how heavy an object is.

Reply:
I'll look into this after the benchmark. I'm not sure how it should look like.